githubmarkdownui/ci.py

```diff
@@ -54,12 +90,54 @@ def ci_task_list(self, status: Optional[CIStatus] = None) -> str:
 
         :param status: Only tasks with the given status will be displayed. If not given, all tasks will be displayed
         """
-        pass
+        # Want to display the results like this, and have the additional info in each line aligned with each other:
+        # <emoji>  <task name> (<task duration>)   <additional info>
+        # Need to find the longest emoji + task name + task duration and use that value to left justify each line.
+        longest_string_length = 0
+        for task in self.tasks:
+            if status and task.metadata.status != status:
+                continue
+
+            string_length = task.get_left_justified_task_string_length()
+
+            if string_length > longest_string_length:
+                longest_string_length = string_length
```

Comment:
Can we make this a property and calculate it only once during initializing an CIJob object?

Reply:
I think it's easier to leave it out as a property, because the `longest_string_length` will differ depending on if you want to filter by `CIStatus`. I could make three class properties for longest success string, longest fail string, and longest string in general but I don't think that's worth it. Also IIUC we'll only be calling `ci_task_list` once per a `CIJob`'s lifespan, so we're not saving any extra work by moving it to a class property

Reply:
There is no guarantee it'll be only called once. But I guess it's fine from a performance perspective. Dropping it for now.